docs/index.rst

```diff
@@ -64,7 +64,7 @@ Contents
    usage
    compilation
    examples
-   generated_api/library_root
+   api/library_root
```

Comment:
This change means that people who have setup non intersphinx links will now direct to invalid URLs.  But the new url makes more sense, as well as `exhale_` is removed from the beginning of every filename as it now requires a subdirectory in the sphinx source dir.  `exhale_` prefix was because it wasn't a strict requirement before.